src/context/Productcontext.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext } from "react";
+import { initialState,productReducer,sortData,filterData } from "./productUtils.js";
+import { products } from "../backend/db/products.js";
+import { useState, useEffect, useReducer } from "react";
+import axios from "axios";
+
+const ProductContext = createContext();
+
+export function useProduct() {
+  return useContext(ProductContext);
+}
+
+export function ProductProvider({ children }) {
+  // const [items, setItems] = useState([]);
+  const [productState,productDispatch] =useReducer(productReducer,initialState);
+
+  const getData = () => {
+    return axios
```

Comment:
Instead of using promise on axios, you can use async-await, trycatch method. Would be preferable